cdk/lib/mparticle-api.ts

```diff
@@ -2,19 +2,36 @@ import { GuApiGatewayWithLambdaByPath } from '@guardian/cdk';
 import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
 import { GuStack } from '@guardian/cdk/lib/constructs/core';
 import { GuLambdaFunction } from '@guardian/cdk/lib/constructs/lambda';
-import { type App, CfnOutput, Duration } from 'aws-cdk-lib';
+import { type App, Duration } from 'aws-cdk-lib';
 import { ComparisonOperator, Metric } from 'aws-cdk-lib/aws-cloudwatch';
-import { Policy, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import {
+	AccountPrincipal,
+	Effect,
+	Policy,
+	PolicyStatement,
+	Role,
+} from 'aws-cdk-lib/aws-iam';
+import { StringParameter } from 'aws-cdk-lib/aws-ssm';
 import { SrLambdaAlarm } from './cdk/sr-lambda-alarm';
 import { SrLambdaDomain } from './cdk/sr-lambda-domain';
 import { nodeVersion } from './node-version';
 
+export interface MParticleApiProps extends GuStackProps {
+	batonAccountIdSSMParam: string;
+}
+
 export class MParticleApi extends GuStack {
-	constructor(scope: App, id: string, props: GuStackProps) {
+	constructor(scope: App, id: string, props: MParticleApiProps) {
 		super(scope, id, props);
 
 		const app = 'mparticle-api';
 
+		const batonAccountId = StringParameter.fromStringParameterName(
+			this,
+			'BatonAccountId',
+			props.batonAccountIdSSMParam,
```

Comment:
that's a good idea to have it in the root of SSM, I didn't realise that was a standard thing.  In alarms-handler it has it's own copy which will be unnecessary.

optional: If it's not already there it could be worth having it as `/accountIds/baton`

Also does it need to be parameterised e.g. for testing or can we just inline it?
```suggestion
			'batonAccountId',
```

Reply:
Done @johnduffell , thanks!